src/azure-cli-core/azure/cli/core/_output.py

```diff
@@ -17,8 +17,13 @@ def __init__(self, cli_ctx=None):
 
     @staticmethod
     def format_yaml(obj):
-        import yaml
-        return yaml.safe_dump(obj.result, default_flow_style=False)
+        from yaml import (safe_dump, representer)
+        import json
+
+        try:
+            return safe_dump(obj.result, default_flow_style=False)
+        except representer.RepresenterError:
+            return safe_dump(json.loads(json.dumps(obj.result)), default_flow_style=False)
```

Comment:
Can you add a comment here to explain the reason like you mentioned in the PR `knack's --query implementation converts the result to an OrderedDict.`?

Reply:
yes will do!